nuclear/extension/NUClearExtension.cmake

```diff
@@ -30,5 +30,9 @@ if(NUCLEAR_LINK_TYPE STREQUAL "SHARED")
   set_target_properties(nuclear_extension PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/lib")
 endif()
 
+# Set warnings
+include(CompilerWarnings)
+set_target_warnings(nuclear_extension)
+
```

Comment:
I feel like these changes would be better to put in the libraries.cmake file in each of these rather than putting them in NUClear Roles directly

Reply:
What benefit does that serve?

Reply:
Means it won't break the 4Tel code when I copy stuff like this back across

Reply:
Would you only be copying extensions across or nuclear roles?

Reply:
Or do you mean all of these changes?

Reply:
If you do mean all of these, does that mean you want a `libraries.cmake` in `shared/message`?